### PR TITLE
Fixes #7618: Add an option to disable display changes graphes if they are slow

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/7618

### CODE REVIEW

**Comment by jooooooon** [comment]
Nice. Thank you very much. :+1

File: rudder-core/src/main/scala/com/normation/rudder/services/reports/NodeChangesService.scala [hunk 1]
```diff
@@ -342,13 +325,10 @@ object NodeChanges {
     d1 isBefore d2
   }
 
-
-
   val data = changesByPeriod(changes).map(a => (a._1,a._2.size))
     JsObj(
         ("x" -> JsArray(data.map(a => Str(a._1))))
-      , ("y" -> JsArray(data.map(a => Num(a._2))))
+      , ("y" -> JsArray(data.map(a => Num(Random.nextInt(10000)))))
```
Comment (ncharles) [inline rudder-core/src/main/scala/com/normation/rudder/services/reports/NodeChangesService.scala]: wait, why a random number here ???

File: rudder-web/src/main/webapp/javascript/rudder/rudder-datatable.js [hunk 2]
```diff
@@ -86,41 +86,46 @@ $.fn.dataTableExt.afnSortData['changes'] = function ( oSettings, iColumn )
...
+          }
+        , axis: {
+              x: {
+                  show : false
+                , type: 'categories'
+              }
+            , y: { show : false }
+          }
+      });
+      recentGraphs[id] = chart
```
Comment (ncharles) [inline rudder-web/src/main/webapp/javascript/rudder/rudder-datatable.js]: most lines don't finish by ; , but certain do
could you be consistent and finish them all by ; ?

File: rudder-web/src/main/webapp/javascript/rudder/rudder-datatable.js [hunk 3]
```diff
@@ -324,7 +329,7 @@ function createRuleTable(gridId, data, needCheckbox, needActions, needCompliance
   columns.push(status);
   if (needCompliance) {
     columns.push(compliance);
-    columns.push(recentChanges);
+    columns.push(recentChanges)
```
Comment (ncharles) [inline rudder-web/src/main/webapp/javascript/rudder/rudder-datatable.js]: why removing the ; ?

File: rudder-core/src/main/scala/com/normation/rudder/services/reports/NodeChangesService.scala (hunk 1 above)
Comment (VinceMacBuche) [inline rudder-core/src/main/scala/com/normation/rudder/services/reports/NodeChangesService.scala]: It was to display good chubby data in my graph (my dev env has 6 changes in 23 rules, most of theml disabled ..)

File: rudder-web/src/main/webapp/javascript/rudder/rudder-datatable.js (hunk 2 above)
Comment (VinceMacBuche) [inline rudder-web/src/main/webapp/javascript/rudder/rudder-datatable.js]: yeah, your right, hate this language

**Comment by VinceMacBuche** [comment]
PR updated

File: rudder-core/src/main/scala/com/normation/rudder/services/reports/NodeChangesService.scala [hunk 4]
```diff
@@ -39,22 +39,17 @@ package com.normation.rudder.services.reports
...
 import com.normation.rudder.domain.reports.ResultRepairedReport
 import com.normation.rudder.repository.CachedRepository
 import com.normation.rudder.repository.ReportsRepository
-
 import net.liftweb.common._
 import net.liftweb.common.Box
 import net.liftweb.common.Loggable
 import net.liftweb.http.js.JE._
-
+import scala.util.Random
```
Comment (ncharles) [inline rudder-core/src/main/scala/com/normation/rudder/services/reports/NodeChangesService.scala]: you can remove this one

**Comment by VinceMacBuche** [comment]
PR updated

**Pull request merged**